zkevm-circuits/src/evm_circuit/util/common_gadget.rs

```diff
@@ -380,75 +380,62 @@ impl<F: Field> TransferToGadget<F> {
         cb: &mut EVMConstraintBuilder<F>,
         receiver_address: WordLoHi<Expression<F>>,
         receiver_exists: Expression<F>,
-        must_create: Expression<F>,
+        opcode_is_create: Expression<F>,
         value: Word32Cell<F>,
         mut reversion_info: Option<&mut ReversionInfo<F>>,
-        account_write: bool,
     ) -> Self {
         let value_is_zero = cb.is_zero_word(&value);
-        if account_write {
-            Self::create_account(
-                cb,
-                receiver_address.clone(),
-                receiver_exists.clone(),
-                must_create.clone(),
-                value_is_zero.expr(),
-                reversion_info.as_deref_mut(),
-            );
-        }
-        let receiver = cb.condition(not::expr(value_is_zero.expr()), |cb| {
-            UpdateBalanceGadget::construct(
-                cb,
-                receiver_address,
-                vec![value.clone()],
-                reversion_info,
-            )
-        });
-
-        Self {
-            receiver,
-            receiver_exists,
-            must_create,
-            value_is_zero,
-        }
-    }
 
-    pub(crate) fn create_account(
-        cb: &mut EVMConstraintBuilder<F>,
-        receiver_address: WordLoHi<Expression<F>>,
-        receiver_exists: Expression<F>,
-        must_create: Expression<F>,
-        value_is_zero: Expression<F>,
-        reversion_info: Option<&mut ReversionInfo<F>>,
-    ) {
+        // Create account
         cb.condition(
             and::expr([
                 not::expr(receiver_exists.expr()),
-                or::expr([not::expr(value_is_zero.expr()), must_create]),
+                or::expr([not::expr(value_is_zero.expr()), opcode_is_create.expr()]),
             ]),
             |cb| {
                 cb.account_write(
                     receiver_address.clone(),
                     AccountFieldTag::CodeHash,
                     cb.empty_code_hash(),
                     WordLoHi::zero(),
-                    reversion_info,
+                    reversion_info.as_deref_mut(),
                 );
             },
         );
+
+        let receiver = cb.condition(not::expr(value_is_zero.expr()), |cb| {
+            cb.increase_balance(receiver_address, value.clone(), reversion_info)
+        });
+
+        Self {
+            receiver,
+            receiver_exists,
+            opcode_is_create,
+            value_is_zero,
+        }
     }
 
     pub(crate) fn assign(
         &self,
         region: &mut CachedRegion<'_, '_, F>,
         offset: usize,
-        (receiver_balance, prev_receiver_balance): (U256, U256),
-        value: U256,
+        rws: &mut StepRws,
+        (receiver_exists, value, opcode_is_create): (bool, U256, bool),
```

Comment:
A bit out-of-scope: do you think it's better we flatten `(receiver_exists, value, opcode_is_create): (bool, U256, bool)` to align with other convention? I believe it's just for avoid `#[allow(clippy::too_many_arguments)]`.

Reply:
I was thinking those three arguments are related to receiver creation.
But I feel it's an overkill to group them in a tuple now. 

Fixed in [f8773e2](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1777/commits/f8773e27f95259a6fe4e589e485a5fb0563746d9)